parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -121,7 +123,7 @@ public void add(T value) {
 
   @Override
   public Metrics metrics() {
-    return ParquetUtil.footerMetrics(writer.getFooter(), metricsConfig);
+    return ParquetUtil.footerMetrics(writer.getFooter(), model.metrics(), inputSchema, metricsConfig);
```

Comment:
We could potentially use the schema from the footer itself in `ParquetUtil` instead of explicitly pass it here; currently have it this way since it's guaranteed that the schema will match.

Reply:
Yeah, as my comment in the `ParquetUtil` class, we should investigate if they are actually the same schema.

Reply:
It should be the same.

Reply:
Thank you for the confirmation!